include/uart.h

```diff
@@ -245,6 +384,157 @@ struct uart_driver_api {
 
 };
 
+#ifdef CONFIG_UART_HIGH_LEVEL_API
+
+/**
+ * @brief Set event handler function.
+ *
+ * @param dev       UART device structure.
+ * @param callback  Event handler.
+ * @param user_data Data to pass to event handler function.
+ *
+ * @retval 0	    If successful, negative errno code otherwise.
+ */
+static inline int uart_callback_set(struct device *dev,
+				    uart_callback_t callback,
+				    void *user_data)
+{
+	const struct uart_driver_api *api =
+			(const struct uart_driver_api *)dev->driver_api;
+
+	return api->callback_set(dev, callback, user_data);
+}
+
+/**
+ * @brief Send given number of bytes from buffer through UART.
+ *
+ * Function returns immediately and event handler,
+ * set using @ref uart_set_callback, is called after transfer is finished.
+ *
+ * @param dev     UART device structure.
+ * @param buf     Pointer to transmit buffer.
+ * @param len     Length of transmit buffer.
+ * @param timeout Timeout in milliseconds.
+ *
+ * @retval enum uart_error In case of error in blocking mode operation.
+ * @retval -EBUSY	   There is already an ongoing transfer.
+ * @retval 0		   If successful, negative errno code otherwise.
+ */
+__syscall int uart_tx(struct device *dev, u8_t *buf, size_t len, u32_t timeout);
+
+static inline int _impl_uart_tx(struct device *dev, u8_t *buf, size_t len,
+				u32_t timeout)
+{
+	const struct uart_driver_api *api =
+			(const struct uart_driver_api *)dev->driver_api;
+
+	return api->tx(dev, buf, len, timeout);
+}
+
+/**
+ * @brief Abort current TX transmission.
+ *
```

Comment:
more clarification needed: what about the buffer. Will `TX_DONE` be triggered (with amount of data sent)?